src/site/content/en/blog/building-a-settings-component/index.md

```diff
@@ -0,0 +1,959 @@
+  left: 50%;
+  background: var(--thumb-highlight-color);
+  transform-origin: center center;            /* goal is a centered scaling circle */
+  transform:                                  /* order here matters!! */
+    translateX(-50%)                          /* counter balances left: 50% */
+    translateY(-50%)                          /* counter balances top: 50% */
+    translateZ(-1px)                          /* PUTS IT BEHIND THE CHECKBOX */
+    scale(var(--thumb-scale))                 /* value we toggle for animation */
+  ;
+  will-change: transform;
+
+  @media (--motionOK) {                       /* transition only if motion is OK */
+    & {
+      transition: transform .2s ease;
+    }
+  }
+}
+
+/* on hover, set scale custom property to "in" state */
+input[type="checkbox"]:hover::before {
+  --thumb-scale: 1;
+}
+```
+
+Creating a circle psuedo element is straightforward work, but **placing it
+behind the element it's attached to**, that's what took work. Here's before and
+after I fixed it:
+
+{% Video 
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/Spdpw5P1MD8ceazneRXo.mp4",
+  className="w-screenshot", 
+  autoplay="true", 
+  loop="true", 
+  muted="true" 
+%}
+
+It's definitely a microinteraction, but important to me to keep the visual
+consistency. The animation scaling technique is the same as we've been using in
+other places, we set a custom property to a new value and let CSS transition it
+based on motion preferences, but it's special feature is `translateZ(-1px)`. The
+parent created a 3D space and this speudo element child tapped into it by
+placing itself slightly back in z-space.
+
+## Accessibility
+
+The YouTube video does a great demonstration of the mouse, keyboard and
+screenreader interactions for this settings component. I'll call out some of the
+details here.
+
+### HTML Element Choices
+
+```html
+<form>
+<header>
+<fieldset>
+<picture>
+<label>
+<input>
+```
+
+Each of these holds hints and tips to the user's browsing tool. Some elements
+provide interaction hints, some connect interactivity, and some help shape the
+acccessibility tree that a screenreader navigates. 
+
+### HTML Attributes
+
+We can hide elements that are not needed by screenreaders, 
+in this case the icon next to the slider and the label:
+
+```html
+<picture aria-hidden="true">
+<label aria-hidden="true">
+```
+
+{% Video 
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/fVjqHRZHQixAaxjeAvDP.mp4",
+  className="w-screenshot", 
+  autoplay="true", 
+  loop="true", 
+  muted="true",
+  width="480px" 
+%}
+
+The above video demonstrates the screenreader flow on Mac OS. Notice how input
+focus is straight from one slider to the next, this is because we've hidden the
+elements that may have otherwise been stops along the way to the next element.
+Without these attributes a user would need to stop, listen and move past the
+picture and duplicative label element.
```

Comment:
```suggestion
The above video demonstrates the screenreader flow on Mac OS. Notice how input
focus moves straight from one slider to the next. This is because we've hidden the
elements that may have become stops on the way to the next element.
Without these attributes a user would need to stop, listen and move past the
picture and duplicate label element.
```